test/API/H5_api_test.c

```diff
@@ -159,28 +245,15 @@ main(int argc, char **argv)
     hid_t       registered_con_id         = H5I_INVALID_HID;
     char       *vol_connector_string_copy = NULL;
     char       *vol_connector_info        = NULL;
-    void       *default_err_data          = NULL;
-    bool        err_occurred              = false;
-    int         chars_written             = 0;
 
     H5open();
 
-    /* Store current error stack printing function since TestInit unsets it */
-    H5Eget_auto2(H5E_DEFAULT, &default_err_func, &default_err_data);
-
     /* Initialize testing framework */
-    if (TestInit(argv[0], usage, NULL, NULL, NULL, 0) < 0) {
-        fprintf(stderr, "Unable to initialize testing framework\n");
-        err_occurred = true;
+    if (TestInit(argv[0], usage, NULL, NULL, NULL, H5_MULTITHREAD_TEST, 0) < 0) {
+        TestErrPrintf("Unable to initialize testing framework\n");
         goto done;
     }
 
-    /* Reset error stack printing function */
-    H5Eset_auto2(H5E_DEFAULT, default_err_func, default_err_data);
-
-    /* Hide all output from testing framework and replace with our own */
-    SetTestVerbosity(VERBO_NONE);
```

Comment:
Instead of overriding output from the testing framework, just use its output now.